packages/perseus/src/widgets/matrix/validate-matrix.ts

```diff
@@ -0,0 +1,22 @@
+import type {PerseusScore} from "../../types";
+import type {
+    PerseusMatrixUserInput,
+    PerseusMatrixValidationData,
+} from "../../validation.types";
+
+/**
+ * Checks user input from the matrix widget to see if it is scorable.
+ *
+ * Note: The matrix widget cannot do any validation without the Scoring
+ * Data because of its use of KhanAnswerTypes as a core part of scoring.
+ *
+ * @see `scoreMatrix()` for more details.
+ */
+function validateMatrix(
```

Comment:
Not going to lose sleep on it, but I feel like we should remove useless validators/scorers. I think it's just noise and at the call site we could just do:

``` TS
const validationError = validator?.(data, userInput)
if (validationError) {
    return beSad()
}
```

Alternatively we could make a `noopValidator` like we have a `noopScorer`, but honestly I'd like to remove `noopScorer` anyway.